crates/circuit/src/dag_circuit.rs

```diff
@@ -1314,28 +1314,21 @@ impl DAGCircuit {
     ///     DAGCircuitError: a creg is not a ClassicalRegister, or is not in
     ///     the circuit.
     #[pyo3(signature = (*cregs))]
-    fn remove_cregs(&mut self, cregs: &Bound<PyTuple>) -> PyResult<()> {
+    fn remove_cregs(&mut self, cregs: Vec<ClassicalRegister>) -> PyResult<()> {
```

Comment:
It seems a bit odd to me for this to consume the `cregs`, since ultimately we remove them by name.

Can this be made to be `&[ClassicalRegister]` or something like that? We could probably make `valid_regs` below be a `Vec<&ClassicalRegister>` I would think.

Reply:
This is a pymethod so I can't use a borrowed type because pyo3 can't do the conversion. To do this I'll have to do as Jake suggests in his comment to split this function into two and have a more efficient rust interface that the python function calls internally.

Reply:
I split this out in: https://github.com/Qiskit/qiskit/pull/14313/commits/9a43895766979f8ee3667248ba0f754f76392cb8